packages/server/src/fhir/operations/asyncjobcancel.ts

```diff
@@ -0,0 +1,44 @@
+import { allOk, assert, badRequest, OperationOutcomeError } from '@medplum/core';
+import { OperationDefinition } from '@medplum/fhirtypes';
+import { Request, Response } from 'express';
+import { asyncWrap } from '../../async';
+import { getAuthenticatedContext } from '../../context';
+import { sendOutcome } from '../outcomes';
+
+export const operation: OperationDefinition = {
+  id: 'AsyncJob-cancel',
+  resourceType: 'OperationDefinition',
+  name: 'asyncjob-cancel',
+  status: 'active',
+  kind: 'operation',
+  code: 'cancel',
+  experimental: true,
+  resource: ['AsyncJob'],
+  system: false,
+  type: false,
+  instance: true,
+  parameter: [{ use: 'out', name: 'return', type: 'OperationOutcome', min: 1, max: '1' }],
+};
+
+/**
+ * Handles HTTP requests for the AsyncJob $cancel operation.
+ * Sets the status of the `AsyncJob` to `cancelled`.
+ */
+export const asyncJobCancelHandler = asyncWrap(async (req: Request, res: Response) => {
+  assert(req.params.id, 'This operation can only be executed on an instance');
+  // Update status of async job
+  try {
+    const { repo } = getAuthenticatedContext();
+    await repo.patchResource('AsyncJob', req.params.id, [
+      { op: 'test', path: '/status', value: 'accepted' },
```

Comment:
I would argue that if the status if already `cancelled`, this operation should be a no-op but return successfully